### ISSUE TITLE
HHVM 3.26 crashing with Xenon enabled

### ISSUE THREAD

**Issue by ssandler** [facebook/hhvm#8221]
### HHVM Version

3.26.3

### Operating System and Version

Ubuntu 16.04

### Standalone code, or other way to reproduce the problem

Set `hhvm.xenon.period=0.997` in `/etc/hhvm/server.ini` and start hhvm (no specific code required).

We're not able to run in repo authoritative mode at this point, it seems like this may not happen in repo authoritative mode based on a conversation with Fred.

### Actual result

HHVM fails to start some of the time and logs this error:
```
[Mon Jun  4 13:17:26 2018] [hphp] [22480:7f9d46323980:0:000001] [] ExternCompiler Error: error reading line: Interrupted system call\n
[Mon Jun  4 13:17:26 2018] [hphp] [22480:7f9d46323980:0:000002] [] ExternCompiler encountered too many communication errors, giving up.
[Mon Jun  4 13:17:26 2018] [hphp] [22485:7f9d46323980:0:000001] [] Lost parent, LightProcess exiting
```

It appears to be a race condition if a Xenon sample happens at the wrong time. 

### Expected result

HHVM starts up and serves requests.

**Comment by Norbyte** [comment]
This is probably the same as #4744.

**Comment by fredemmott** [comment]
Should be addressed by c7622408c8b3a6f861ae9292433623163966cd21
